functions/New-DbaAgentJob.ps1

```diff
@@ -0,0 +1,476 @@
+        [ValidateScript({
+            if(($EmailLevel -in 1, "OnSuccess", 2, "OnFailure", 3, "Always") -and ($_.Length -ge 1))
+            {
+                $true
+            }
+            else
+            {
+                Throw "Please set the e-mail operator when the e-mail level parameter is set."
+            }
+        })]
+        [string]$EmailOperatorName,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($NetsendLevel -in 1, "OnSuccess", 2, "OnFailure", 3, "Always") -and ($_.Length -ge 1))
+            {
+                $true
+            }
+            else
+            {
+                Throw "Please set the netsend operator when the netsend level parameter is set."
+            }
+        })]
+        [string]$NetsendOperatorName,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($PageLevel -in 1, "OnSuccess", 2, "OnFailure", 3, "Always") -and ($_.Length -ge 1))
+            {
+                $true
+            }
+            else
+            {
+                Throw "Please set the page operator when the page level parameter is set."
+            }
+        })]
+        [string]$PageOperatorName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$DeleteLevel = '',
+        [bool]$Force,
+        [switch]$Silent
+    )
+
+    BEGIN
+    {
+        # Check the input for the job id or job name
+        if($JobName.Length -lt 1)
+        {
+            Stop-Function -Message "Please enter a job name." -Silent $Silent -InnerErrorRecord $_ -Target $JobName
+            return
+        }
+
+        # Check of the event log level is of type string and set the integer value
+        if(($EmailLevel.GetType().Name -eq 'String') -and ($EventLogLevel.Length -ge 1))
+        {
+            switch ($EventLogLevel)
+            {
+                {($_ -eq "Never") -or ($_ -eq '')} { $EventLogLevel = 0 } 
+                "OnSuccess" { $EventLogLevel = 1 } 
+                "OnFailure" { $EventLogLevel = 2 }
+                "Always" { $EventLogLevel = 3 } 
+            }
+        }
+
+        # Check of the email level is of type string and set the integer value
+        if(($EmailLevel.GetType().Name -eq 'String') -and ($EmailLevel.Length -ge 1))
+        {
+            switch ($EmailLevel)
+            {
+                {($_ -eq "Never") -or ($_ -eq '')} { $EmailLevel = 0 } 
+                "OnSuccess" { $EmailLevel = 1 } 
+                "OnFailure" { $EmailLevel = 2 }
+                "Always" { $EmailLevel = 3 } 
+            }
+        }
+
+        # Check of the net send level is of type string and set the integer value
+        if(($NetsendLevel.GetType().Name -eq 'String') -and ($NetsendLevel.Length -ge 1))
+        {
+            switch ($NetsendLevel)
+            {
+                {($_ -eq "Never") -or ($_ -eq '')} { $NetsendLevel = 0 } 
```

Comment:
I think this would be better as the final entry:
default { 0 }

also:
$netsendlevel = switch($netsendlevel) { 'OnSuccess' { 1 } 'Onfailure' {2} 'Always' { 3 } default {0}}
will replace those 8 lines

same for the other switches